### PR TITLE
[stable/apm-server] Add hpa for apm server when kind is deployment

### PR DESCRIPTION
**Pull request by 759803573**
#### What this PR does / why we need it:
Added HPA for apm-server when kind is deployment.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: stable/apm-server/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    app: {{ template "apm-server.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ template "apm-server.fullname" . }}
+  minReplicas: {{ .Values.autoscaling.minReplicas }}
```
Comment (mumoshu) [inline stable/apm-server/templates/hpa.yaml]: Can we probably simplify this like the below?

```
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: {{ template "apm-server.fullname" . }}
{{ (unset .Values.autoscaling "enabled") | toYaml | indent 4 }}
```

Comment (759803573) [reply stable/apm-server/templates/hpa.yaml]: Thanks for tips and have modified

**Comment by mumoshu** [comment]
OMG! Sorry for the delayed response.

**Comment by 759803573** [comment]
hi @mumoshu  
 The pr is closed. 
  How should I reopen it? Do I need to resubmit a pr? 
thanks

**Comment by mumoshu** [comment]
@759803573 Probably that's the case.. Would you mind resubmitting it? As I have the priv to run /ok-to-test now, I'll be able to handle this until merged now.

Perhaps you'll get a conflict in `Chart.yaml` `version` as I think I've recently merged another commit that changes it.

**Comment by 759803573** [comment]
@mumoshu  I don't mind resubmitting. The new pr is  #14377. (Sorry, my English is very poor.)

**Pull request closed without merging**